### PR TITLE
handle SVG anchor tags

### PR DESCRIPTION
**Pull request by cherifGsoul**
For #81 

### The problem 
SVG anchor tags doesn't have `pathname` property so using it doesn't work with pushstate.

#### The changes:
Check if the `node`  has `namespaceURI` property equal to `http://www.w3.org/2000/svg`  if so the pathanme will be `xlink:href` or `href` attribute value and use it to update the `window.location`

### CODE REVIEW

File: can-route-pushstate-iframe-test.js [hunk 1]
```diff
@@ -767,5 +767,69 @@ function makeTest(mapModuleName){
...
+				var a = win.document.createElementNS("http://www.w3.org/2000/svg", "a");
+				a.setAttributeNS("http://www.w3.org/2000/svg", "href", "/books/40");
+				a.innerHTML = "Click Me";
+				win.document.body.appendChild(a);
+				domEvents.dispatch(a, "click");
+				setTimeout(function() {
+					equal(win.location.pathname, '/books/40');
+					start();
+					done();
+				}, 30);
```
Comment (justinbmeyer) [inline can-route-pushstate-iframe-test.js]: these timeouts often cause problems

File: can-route-pushstate.js [hunk 2]
```diff
@@ -159,41 +159,59 @@ canReflect.assign(PushstateObservable.prototype, {
 
 			// If link is within the same domain and descendant of `root`.
 			if (window.location.host === linksHost) {
-				var root = cleanRoot();
-
+				var root = cleanRoot(),
```
Comment (justinbmeyer) [inline can-route-pushstate.js]: Is it possible to pull out `anchorClickHandler` (or the important part of it) so it is easier to test directly?  If we can avoid creating iframe based tests, we should do that.

Comment (cherifGsoul) [reply can-route-pushstate.js]: Will do!

Comment (cherifGsoul) [reply can-route-pushstate.js]: @justinbmeyer I changed the function, can you please review?

File: can-route-pushstate.js [hunk 3]
```diff
@@ -159,23 +161,31 @@ canReflect.assign(PushstateObservable.prototype, {
...
+					nodePathWithSearch = href;
+				}
 
 				// If the link is within the `root`.
-				if (node.pathname.indexOf(root) === 0) {
-
-					// Removes root from url.
-					var nodePathWithSearch = node.pathname + node.search,
-						url = nodePathWithSearch.substr(root.length);
+				if (pathname.indexOf(root) === 0) {
```
Comment (chasenlehara) [inline can-route-pushstate.js]: I think this should check for `pathname !== undefined` first in case the two cases above aren’t hit.

Comment (cherifGsoul) [reply can-route-pushstate.js]: Done!

**Comment by matthewp** [comment]
lgtm, merging.

**Pull request merged**